### ISSUE TITLE
no-unused-prop-types fails on readonly flow types

### ISSUE THREAD

**Issue by TheSavior** [yannickcr/eslint-plugin-react#1388]
If your Props type definition is:

```
type Props = {
  +prop1: ?number,
  +prop2: number,
};
```

The rule will provide the warnings:

```
warning  '+' PropType is defined but prop is never used  react/no-unused-prop-types
```

**Comment by ljharb** [comment]
What version of flow are you using?

**Comment by TheSavior** [comment]
```
$ flow version
Flow, a static type checker for JavaScript, version 0.53.1
```

**Comment by jseminck** [comment]
Confirmed the bug. Will fix it!

**Comment by TheSavior** [comment]
I'm not sure, I'll run this rule through our codebase and flag all of the places that it seems to have false positives. That should give a pretty good signal.

**Comment by ljharb** [comment]
@jseminck there's covariant (+) and contravariant (-), at least
